packages/wonder-blocks-core/components/clickable-behavior.js

```diff
@@ -269,13 +298,81 @@ export default class ClickableBehavior extends React.Component<
         this.dragging = false;
     }
 
+    async runCallbackAndMaybeNavigate(e: SyntheticEvent<>) {
+        const {
+            onClick = undefined,
+            beforeNav = undefined,
+            safeWithNav = undefined,
+            skipClientNav,
+            history,
+        } = this.props;
+        let shouldNavigate = true;
+
+        if (onClick) {
+            onClick(e);
+        }
+
+        // If onClick() has called e.preventDefault() then we shouldn't
+        // navigate.
+        if (e.defaultPrevented) {
+            shouldNavigate = false;
+        }
+
+        // Prevent navigation.
+        e.preventDefault();
+
+        try {
+            if (beforeNav) {
+                this.setState({waiting: true});
+                await beforeNav();
+            }
+
+            if (safeWithNav) {
+                if (history && !skipClientNav) {
+                    if (!this.state.waiting) {
+                        // We only show the spinner for safeWithNav when doing
+                        // a server-side navigation since since the spinner is
+                        // indicating that we're waiting for navigation to occur.
+                        this.setState({waiting: true});
+                    }
```

Comment:
I think this `if` block is in the wrong branch of `if (history && !skipClientNav) {`, is that possible? If not, I think the comment is confusing.

Reply:
Good catch.  Yes, the `setState` should go down below with the `await safeWithNav()`.